src/ape/utils/misc.py

```diff
@@ -490,9 +490,33 @@ def is_zero_hex(address: str) -> bool:
         return False
 
 
+def dict_defaults(mapping: Dict[str, Any], defaults: Dict[str, Any], depth=0):
+    """Conditionally overlay given default on a dict if no value is set"""
+    for key, value in defaults.items():
+        if key not in mapping:
+            mapping[key] = value
+        elif isinstance(value, dict):
+            dict_defaults(mapping[key], value, depth + 1)
+    return mapping
+
+
+def dict_overlay(mapping: Dict[str, Any], overlay: Dict[str, Any], depth=0):
```

Comment:
```suggestion
def dict_overlay(mapping: Dict[str, Any], overlay: Dict[str, Any], depth: int=0):
```

Reply:
I thought it was customary to leave off the type hint when you have an explicit non-None default.  Though I'm not sure where I got that from.  Might be mixing that up with Rust or something.  :shrug: